src/protocols/tulip.ts

```diff
@@ -332,146 +356,442 @@ export class ProtocolTulip implements IProtocolMoneyMarket, IProtocolVault {
+            isWritable: true,
+          }, // 16
+          {
+            pubkey: orcaVault.farmData.poolSwapTokenB.address,
+            isSigner: false,
+            isWritable: true,
+          }, // 17
+          {
+            pubkey: orcaVault.farmData.globalFarm,
+            isSigner: false,
+            isWritable: true,
+          }, // 18
+          {
+            pubkey: orcaVault.farmData.userFarmAddr,
+            isSigner: false,
+            isWritable: true,
+          }, // 19
+          {
+            pubkey: orcaVault.farmData.convertAuthority,
+            isSigner: false,
+            isWritable: false,
+          }, // 20
+          {
+            pubkey: orcaVault.farmData.poolSwapAccount,
+            isSigner: false,
+            isWritable: true,
+          }, // 21
+          {
+            pubkey: orcaVault.farmData.poolSwapAuthority,
+            isSigner: false,
+            isWritable: false,
+          }, // 22
+          {
+            pubkey: orcaVault.farmData.swapPoolMint.address,
+            isSigner: false,
+            isWritable: true,
+          }, // 23
+          {
+            pubkey: orcaVault.farmData.farmTokenMint,
+            isSigner: false,
+            isWritable: true,
+          }, // 24
+          {
+            pubkey: orcaVault.shareMint,
+            isSigner: false,
+            isWritable: true,
+          }, // 25
+          {
+            pubkey: orcaVault.farmData.swapPoolFeeTokenAccount,
+            isSigner: false,
+            isWritable: true,
+          }, // 26
+          { pubkey: TOKEN_PROGRAM_ID, isSigner: false, isWritable: false }, // 27
+          { pubkey: orca.ORCA_POOL_PROGRAM_ID, isSigner: false, isWritable: false }, // 28
+          {
+            pubkey: orca.ORCA_FARM_PROGRAM_ID,
+            isSigner: false,
+            isWritable: false,
+          }, // 29
+          { pubkey: ephemeralTrackingAccount, isSigner: false, isWritable: true }, // 30
+          { pubkey: anchor.web3.SystemProgram.programId, isSigner: false, isWritable: false }, // 31
+          { pubkey: orcaVault.farmData.feeCollectorTokenAccount, isSigner: false, isWritable: true } // 32
+        );
+        break;
+      case tulip.VaultType.OrcaDD:
+        const orcaDDVault = vaultInfo as tulip.OrcaDDVaultInfo;
+        const userTokenAccountA = await getAssociatedTokenAddress(orcaDDVault.ddFarmData.tokenAMint, userKey);
+        preInstructions.push(await createATAWithoutCheckIx(userKey, orcaDDVault.ddFarmData.tokenAMint));
+        const userTokenAccountB = await getAssociatedTokenAddress(orcaDDVault.ddFarmData.tokenBMint, userKey);
+        preInstructions.push(await createATAWithoutCheckIx(userKey, orcaDDVault.ddFarmData.tokenBMint));
+        const userFarmTokenAccount = await getAssociatedTokenAddress(orcaDDVault.farmData.farmTokenMint, userKey);
+        preInstructions.push(await createATAWithoutCheckIx(userKey, orcaDDVault.farmData.farmTokenMint));
+        preInstructions.push(
```

Comment:
Add some descriptive comment for the intention of putting `withdrawTrackingAccountIx` in preInstructions